sdk/src/test/scala/coop/rchain/sdk/merging/DagMergingLogicSpec.scala

```diff
@@ -0,0 +1,193 @@
+package coop.rchain.sdk.merging
+import coop.rchain.sdk.dag.merging.DagMergingLogic._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.scalacheck.Checkers
+import cats.syntax.all._
+
+class DagMergingLogicSpec extends AnyFlatSpec with Matchers with Checkers {
+  "withDependencies" should "emit element and all dependencies" in {
+    val dependentsMap = Map(1 -> Set(3, 9), 3 -> Set(5), 5 -> Set(6), 4 -> Set(6))
+    val rejects       = withDependencies(Set(1), dependentsMap)
+    rejects shouldBe Set(1, 3, 9, 5, 6)
+  }
+
+  "incompatibleWithFinal" should "output conflicting with finally accepted or depending on finally rejected" in {
+    val acceptedFinally = Set(1, 2)
+    val rejectedFinally = Set(5, 6)
+    val conflictsMap    = Map(1 -> Set(11, 12), 2 -> Set(21, 22), 3 -> Set(31, 32))
+    val dependencyMap   = Map(5 -> Set(51, 52), 6 -> Set(61, 62), 7 -> Set(71, 72))
+    val r               = incompatibleWithFinal(acceptedFinally, rejectedFinally, conflictsMap, dependencyMap)
+    r shouldBe Set(11, 12, 21, 22, 51, 52, 61, 62)
+  }
+
+  "partitionScope" should "output non intersecting partitions" in {
+    val views = Seq(Set(1, 2, 3, 4), Set(4, 5, 6, 7), Set(7, 8, 9), Set(9, 10))
+    val r     = partitionScope(views)
+    r shouldBe Seq(Set(1, 2, 3, 4), Set(5, 6, 7), Set(8, 9), Set(10))
+  }
+
+  "computeConflictsMap" should "compute correct map." in {
```

Comment:
What does it mean _correct map_. Please try to describe the purpose of the test.